src/uct/ib/mlx5/ib_mlx5_dv.c

```diff
@@ -299,8 +299,11 @@ static ucs_status_t uct_ib_mlx5dv_md_open(struct ibv_device *ibv_device,
             dev->ext_atomic_arg_sizes_be = arg_size;
         }
 
-        if (UCT_IB_MLX5DV_GET(atomic_caps, cap, fetch_add_pci_atomic)) {
+        if (UCT_IB_MLX5DV_GET(atomic_caps, cap, fetch_add_pci_atomic) ||
+            UCT_IB_MLX5DV_GET(atomic_caps, cap, compare_swap_pci_atomic)) {
             dev->flags |= UCT_IB_DEVICE_FLAG_PCI_ATOMICS;
+            dev->pci_fadd_arg_sizes = UCT_IB_MLX5DV_GET(atomic_caps, cap, fetch_add_pci_atomic) << 2;
```

Comment:
can we remove UCT_IB_DEVICE_FLAG_PCI_ATOMICS now? and rely on (dev->pci_fadd_arg_sizes || dev->pci_cswap_arg_sizes)?